Sources/Orbit/Support/TextFields/TextField.swift

```diff
@@ -200,17 +184,27 @@ public struct TextField: UIViewRepresentable, TextFieldBuildable {
             switch result {
                 case .accept:
                     // Accept the proposed change
-                    textFieldValue = text.replacingCharacters(in: range, with: string)
+                    let newValue = text.replacingCharacters(in: range, with: string)
+                    replaceValue(with: newValue)
                     return true
-                case .replace(let modifiedTextFieldValue):
-                    // Refuse the proposed change, replace the textfield with modified value
-                    textField.text = modifiedTextFieldValue
-                    textFieldValue = modifiedTextFieldValue
+                case .replace(let modifiedValue):
+                    // Refuse the proposed change, replace the text with modified value
+                    textField.text = modifiedValue
+                    replaceValue(with: modifiedValue)
                     return false
                 case .reject:
                     return false
             }
         }
+
+        func replaceValue(with text: String) {
+            textFieldValue = text
+
+            DispatchQueue.main.async { [weak self] in
```

Comment:
Main thread dispatch isn't retained anywhere so it should be safe to go with strong reference. 👍

Reply:
`Task { @MainActor` is also an option...

Reply:
Updated with `Task { @MainActor` in 3 places where we need to dispatch the update to prevent the realtime SwiftUI warning "Modifying state during view update, this will cause undefined behavior"